doc/flame/examples/lib/navigator.dart

```diff
@@ -0,0 +1,412 @@
+    _borderPaint = Paint()
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = 2
+      ..color = borderColor;
+  }
+
+  final String text;
+  final void Function() action;
+  final TextPainter _textDrawable;
+  late final Offset _textOffset;
+  late final RRect _rrect;
+  late final Paint _borderPaint;
+  late final Paint _bgPaint;
+
+  @override
+  void render(Canvas canvas) {
+    canvas.drawRRect(_rrect, _bgPaint);
+    canvas.drawRRect(_rrect, _borderPaint);
+    _textDrawable.paint(canvas, _textOffset);
+  }
+
+  @override
+  void onTapDown(TapDownEvent event) {
+    scale = Vector2.all(1.05);
+  }
+
+  @override
+  void onTapUp(TapUpEvent event) {
+    scale = Vector2.all(1.0);
+    action();
+  }
+
+  @override
+  void onTapCancel(TapCancelEvent event) {
+    scale = Vector2.all(1.0);
+  }
+}
+
+abstract class SimpleButton extends PositionComponent with TapCallbacks {
+  SimpleButton(this._iconPath, {super.position}) : super(size: Vector2.all(40));
+
+  final Paint _borderPaint = Paint()
+    ..style = PaintingStyle.stroke
+    ..color = const Color(0x66ffffff);
+  final Paint _iconPaint = Paint()
+    ..style = PaintingStyle.stroke
+    ..color = const Color(0xffaaaaaa)
+    ..strokeWidth = 7;
+  final Path _iconPath;
+
+  void action();
+
+  @override
+  void render(Canvas canvas) {
+    canvas.drawRRect(
+      RRect.fromRectAndRadius(size.toRect(), const Radius.circular(8)),
+      _borderPaint,
+    );
+    canvas.drawPath(_iconPath, _iconPaint);
+  }
+
+  @override
+  void onTapDown(TapDownEvent event) {
+    _iconPaint.color = const Color(0xffffffff);
+  }
+
+  @override
+  void onTapUp(TapUpEvent event) {
+    _iconPaint.color = const Color(0xffaaaaaa);
+    action();
+  }
+
+  @override
+  void onTapCancel(TapCancelEvent event) {
+    _iconPaint.color = const Color(0xffaaaaaa);
+  }
+}
+
+class BackButton extends SimpleButton with HasGameRef<NavigatorGame> {
+  BackButton()
+      : super(
+          Path()
+            ..moveTo(22, 8)
+            ..lineTo(10, 20)
+            ..lineTo(22, 32)
+            ..moveTo(12, 20)
+            ..lineTo(34, 20),
+          position: Vector2.all(10),
+        );
+
+  @override
+  void action() => gameRef.navigator.popPage();
+}
+
+class PauseButton extends SimpleButton with HasGameRef<NavigatorGame> {
+  PauseButton()
+      : super(
+          Path()
+            ..moveTo(14, 10)
+            ..lineTo(14, 30)
+            ..moveTo(26, 10)
+            ..lineTo(26, 30),
+          position: Vector2(60, 10),
+        );
+  @override
+  void action() => gameRef.navigator.showPage('pause');
+}
+
+class LevelPage extends Page {
+  LevelPage({super.builder});
+
+  @override
+  void onActivate() {
+    resumeTime();
+    removeRenderEffect();
+  }
+
+  @override
+  void onDeactivate() {
+    stopTime();
+    addRenderEffect(
+      PaintRenderEffect()
+        ..addDesaturation(opacity: 0.5)
+        ..addBlur(3.0),
+    );
+  }
+}
+
+class Level1PageImpl extends Component {
```

Comment:
Why the `Impl`?

Reply:
Currently, `Page` is the component whose job is to build the actual contents of the page when it is shown. So, I wanted to make a distinction that `SomethingPage` is a subclass of `Page`, and its content would then be `SomethingPageImpl`, or `SomethingScreen`, or `SomethingComponent`, or `SomethingPageContent`, etc.

The `Page` is an equivalent to Flutter's `Route`. Perhaps if we renamed it into `Route` it would be less confusing?

Reply:
Renamed Page into Route now.

Reply:
You forgot to rename the components here though, because I guess now the components should be have just the `Page` suffix and not the `PageImpl` one?

Reply:
They are called `Level1Page` and `Level2Page` now, maybe you're looking at the older version of code?

Reply:
I was indeed, must have continued on an old tab. :sweat_smile: